<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=UTF-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<title>发表论文</title>
<!-- MathJax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML' async>
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
	  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
<!-- End MathJax -->
</head>
<body>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">序列比对</div>
<div class="menu-item"><a href="index.html">首页</a></div>
<div class="menu-item"><a href="psa.html">双序列比对</a></div>
<div class="menu-item"><a href="msa.html">多序列比对</a></div>
<div class="menu-item"><a href="measure.html">评分标准</a></div>
<div class="menu-item"><a href="mga.html">基因组比对</a></div>
<div class="menu-item"><a href="realignment.html">Realignment</a></div>
<div class="menu-item"><a href="datasets.html">测试数据集</a></div>
<div class="menu-item"><a href="simu_data.html">生成模拟数据</a></div>
<div class="menu-item"><a href="tools.html">实用工具</a></div>
<div class="menu-item"><a href="modules.html">实用开发模块</a></div>
<div class="menu-item"><a href="software_notes.html">软件使用</a></div>
<div class="menu-item"><a href="papers.html" class="current">发表论文</a></div>
<div class="menu-item"><a href="https://github.com/malabz" target="blank">Github</a></div>
</td>
<td id="layout-content">
<div id="toptitle">
<h1>发表论文</h1>
</div>
<h2><a href="http://lab.malab.cn/soft/halign" target=&ldquo;blank&rdquo;>HAlign</a></h2>
<p>1. <a href="https://academic.oup.com/bioinformatics/article/31/15/2475/188425" target=&ldquo;blank&rdquo;><b>HAlign</b></a>: Fast multiple similar DNA/RNA sequence alignment based on the centre star strategy. <b>Bioinformatics</b>. 2015 (<a href="https://github.com/malabz/HAlign" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>2. <a href="https://link.springer.com/content/pdf/10.1186/s13015-017-0116-x.pdf" target=&ldquo;blank&rdquo;><b>HAlign 2</b></a>: efficient ultra-large multiple sequence alignment and phylogenetic tree reconstruction with distributed and parallel computing. <b>Algorithms for Molecular Biology</b>. 2017 (<a href="https://github.com/malabz/HAlign" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>3. <a href="https://doi.org/10.1093/molbev/msac166" target=&ldquo;blank&rdquo;><b>HAlign 3</b></a>: fast multiple alignment of ultra-large numbers of similar DNA/RNA sequences. <b>Molecular Biology and Evolution</b>. 2022 (<a href="https://github.com/malabz/HAlign-3" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>4. <a href="https://academic.oup.com/bioinformatics/article/40/12/btae718/7912339" target=&ldquo;blank&rdquo;><b>HAlign 4</b></a>: a new strategy for rapidly aligning millions of sequences. <b>Bioinformatics</b>. 2024 (<a href="https://github.com/malabz/HAlign-4" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>FMAlign</h2>
<p>1. <a href="https://academic.oup.com/bib/article/23/1/bbab519/6458932" target=&ldquo;blank&rdquo;><b>FMAlign</b></a>: A novel fast multiple nucleotide sequence alignment method based on FM-index. <b>Briefings in Bioinformatics</b>. 2021
</p>
<p>2. <a href="https://doi.org/10.1093/bioinformatics/btae014" target=&ldquo;blank&rdquo;><b>FMAlign2</b></a>: a novel fast multiple nucleotide sequence alignment method for ultralong datasets. <b>Bioinformatics</b>. 2024 (<a href="https://github.com/malabz/FMAlign2" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>MSA</h2>
<p>1. <a href="https://pubmed.ncbi.nlm.nih.gov/28646874/" target=&ldquo;blank&rdquo;><b>CMSA</b></a>: a heterogeneous CPU/GPU computing system for multiple similar RNA/DNA sequence alignment. <b>BMC Bioinformatics</b>. 2017 (<a href="https://github.com/wangvsa/CMSA" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>2. <a href="https://link.springer.com/article/10.1186/s12918-017-0476-3" target=&ldquo;blank&rdquo;>Reconstructing evolutionary trees in parallel for massive sequences</a>. <b>BMC Systems Biology</b>. 2017
</p>
<p>3. <a href="https://www.liebertpub.com/doi/10.1089/cmb.2017.0040" target=&ldquo;blank&rdquo;>Multiple Sequence Alignment Based on a Suffix Tree and Center-Star Strategy</a>: A Linear Method for Multiple Nucleotide Sequence Alignment on Spark Parallel Framework. <b>Journal of Computational Biology</b>. 2017 (<a href="https://github.com/suwenhecn/MASC" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>4. <a href="http://www.jgenetgenomics.org/en/article/doi/10.1016/j.jgg.2018.07.004" target=&ldquo;blank&rdquo;><b>VCSRA</b></a>: A fast and accurate multiple sequence alignment algorithm with a high degree of parallelism. <b>Journal of Genetics and Genomics</b>. 2018 (<a href="https://github.com/DongDongNeophyte/VCSRA_code" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>5. <a href="http://www.juestc.uestc.edu.cn/cn/article/doi/10.12178/1001-0548.2021406" target=&ldquo;blank&rdquo;>Improved Center Star Alignment Algorithm Based on Profile Alignment</a>. <b>Journal of University of Electronic Science and Technology of China</b>. 2022
</p>
<p>6. <a href="https://doi.org/10.1093/bioinformatics/btac658" target=&ldquo;blank&rdquo;><b>WMSA</b></a>: a novel method for multiple sequence alignment of DNA sequences. <b>Bioinformatics</b>. 2022 (<a href="https://github.com/malabz/WMSA/" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>7. <a href="https://doi.org/10.1093/bib/bbad190" target=&ldquo;blank&rdquo;><b>WMSA2</b></a>: a multiple DNA/RNA sequence alignment tool implemented with accurate progressive mode and a fast win-win mode combining the center star and progressive strategies. <b>Briefings in Bioinformatics</b>. 2023 (<a href="https://github.com/malabz/WMSA2/" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>MSA post-processing</h2>
<p>1. <a href="https://doi.org/10.1371/journal.pcbi.1011988" target=&ldquo;blank&rdquo;><b>TPMA</b></a>: A two pointers meta-alignment tool to ensemble different multiple nucleic acid sequence alignments. <b>PLOS COMPUTATIONAL BIOLOGY</b>. 2024 (<a href="https://github.com/malabz/TPMA" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>2. <a href="https://academic.oup.com/nargab/article/6/4/lqae170/7928182" target=&ldquo;blank&rdquo;><b>ReAlign-N</b></a>: an integrated realignment approach for multiple nucleic acid sequence alignment, combining global and local realignments. <b>NAR Genomics and Bioinformatics</b>. 2024 (<a href="https://github.com/malabz/ReAlign-N" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>3. <a href="https://link.springer.com/chapter/10.1007/978-981-95-0030-7_38" target=&ldquo;blank&rdquo;><b>ReAlign-Star</b></a>: an optimized realignment method for multiple sequence alignment, targeting star algorithm tools. <b>International Conference on Intelligent Computing</b>. 2025 (<a href="https://github.com/malabz/ReAlign-Star" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<p>4. <a href="https://academic.oup.com/bioinformatics/article/41/8/btaf421/8213638?login=false" target=&ldquo;blank&rdquo;><b>ReAlign-P</b></a>: a vertical iterative realignment method for protein multiple sequence alignment. <b>Bioinformatics</b>. 2025 (<a href="https://github.com/malabz/ReAlign-P" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>PSA</h2>
<p>1. <a href="https://doi.org/10.1093/bib/bbaf061" target=&ldquo;blank&rdquo;><b>FORAlign</b></a>: Accelerating gap-affine DNA pairwise sequence alignment using FOR-blocks based on FOur Russians approach with linear space complexity. <b>Briefings in Bioinformatics</b>. 2025 (<a href="https://github.com/malabz/FORAlign" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>PGA</h2>
<p>1. <a href="https://doi.org/10.1101/gr.279763.124" target=&ldquo;blank&rdquo;>Fast sequence alignment for centromere with RaMA</a>. <b>Genome Research</b>. 2025 (<a href="https://github.com/metaphysicser/RaMA" target=&ldquo;blank&rdquo;>soft</a>)
</p>
<h2>Review</h2>
<p>1. <a href="https://www.mdpi.com/2218-273X/12/4/546" target=&ldquo;blank&rdquo;>Developments in Algorithms for Sequence Alignment</a>: A Review. <b>Biomolecules</b>. 2022
</p>
<p>2. <a href="https://doi.org/10.1093/bib/bbac069" target=&ldquo;blank&rdquo;>A survey on the algorithm and development of multiple sequence alignment</a>. <b>Briefings in Bioinformatics</b>. 2022
</p>
<p>3. Advances in post-processing methods for multiple sequence alignment. <b>Frontiers in Genetics</b>. 2025
</p>
</td>
</tr>
</table>
</body>
</html>
