<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=UTF-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<title><a href="realignment.html" target=&ldquo;blank&rdquo;>Realignment Methods</a></title>
<!-- MathJax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML' async>
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
	  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
<!-- End MathJax -->
</head>
<body>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">序列比对</div>
<div class="menu-item"><a href="index.html">首页</a></div>
<div class="menu-item"><a href="psa.html">双序列比对</a></div>
<div class="menu-item"><a href="msa.html">多序列比对</a></div>
<div class="menu-item"><a href="mga.html">基因组比对</a></div>
<div class="menu-item"><a href="realignment.html" class="current">重比对</a></div>
<div class="menu-item"><a href="measure.html">评分标准</a></div>
<div class="menu-item"><a href="datasets.html">测试数据集</a></div>
<div class="menu-item"><a href="simu_data.html">生成模拟数据</a></div>
<div class="menu-item"><a href="tools.html">实用工具</a></div>
<div class="menu-item"><a href="modules.html">实用开发模块</a></div>
<div class="menu-item"><a href="software_conclu.html">发表软件使用指南</a></div>
<div class="menu-item"><a href="software_notes.html">第三方软件避坑技巧</a></div>
<div class="menu-item"><a href="papers.html">发表论文</a></div>
<div class="menu-item"><a href="https://github.com/malabz" target="blank">Github</a></div>
</td>
<td id="layout-content">
<div id="toptitle">
<h1><a href="realignment.html" target=&ldquo;blank&rdquo;>Realignment Methods</a></h1>
</div>
<h2>Realigners based on vertical and horizontal partitioning</h2>
<h3><a href="https://academic.oup.com/bioinformatics/article-abstract/19/9/1155/285019" target=&ldquo;blank&rdquo;>1. RASCAL (2003)</a></h3>
<p>The alignment is initially divided <b>horizontally and vertically</b> to identify well-aligned, reliable regions. Potential alignment errors are detected by comparing statistical models of the reliable regions. RASCAL then performs a single realignment of each badly aligned region using an algorithm similar to that implemented in ClustalW.
</p>
<p>Partitioning steps:
</p>
<ol>
<li><p>The alignment is divided horizontally into sequence subfamilies using Secator
</p>
</li>
<li><p>Then the alignment is divided vertically into 'core block&rsquo; regions that are reliably aligned in the majority of the sequences.
</p>
</li>
</ol>
<p>These global core blocks are determined using the mean distance column scores implemented in the NorMD OF. Local core block regions are also determined for each subfamily individually, using the same method as for the complete family.
</p>
<ul>
<li><p>Test data: Protein (BALIBASE, ProDom)
</p>
</li>
<li><p><a href="ftp://ftp-igbmc.u-strasbg.fr/pub/RASCAL" target=&ldquo;blank&rdquo;>The source code link is unavailable.</a>
</p>
</li>
</ul>
<h3><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-144#Sec3" target=&ldquo;blank&rdquo;>2. Crumble and Prune (2011)</a></h3>
<p>[vertical] <b>Crumble</b> deals with problems that are large in sequence length. It breaks up long alignment problems into shorter problems. 
</p>
<p>[horizontal] <b>Prune</b> handles problems with a large number of sequences. It cuts up deep alignment problems into sub-problems with fewer sequences. 
</p>
<p><b>Job-tree</b> combine Crumble and Prune to align long and deep sequences.
</p>
<ul>
<li><p>Test data: human genome, Rfam 
</p>
</li>
<li><p><a href="http://hgwdev.cse.ucsc.edu/~krish/crumble_prune/" target=&ldquo;blank&rdquo;>The source code link is unavailable.</a>
</p>
</li>
</ul>
<h2><a href="realigner_ver.html" target=&ldquo;blank&rdquo;>Realigners based on vertical partitioning</a></h2>
<h2><a href="realigner_hor.html" target=&ldquo;blank&rdquo;>Realigners based on horizontal partitioning</a></h2>
<h2>Realignment in MSA tools</h2>
<h3><a href="realign_msa_ver.html" target=&ldquo;blank&rdquo;>Vertical-oriented realignment in MSA tools</a></h3>
<h3><a href="realign_msa_hor.html" target=&ldquo;blank&rdquo;>Horizontal-oriented realignment in MSA tools</a></h3>
</td>
</tr>
</table>
</body>
</html>
